triggers/astarte_trigger_test.go

```diff
@@ -0,0 +1,707 @@
+		},
+		"simple_triggers": [
+		  {
+			"type": "data_trigger",
+			"on": "incoming_data",
+			"interface_name": "AAA",
+			"interface_major": "2",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggermajornotnull))
+	if err != nil {
+		t.Error("This trigger should have passed!")
+
+	}
+	DeviceTriggermajornullable := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		  {
+			"type": "data_trigger",
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggermajornullable))
+	if err != nil {
+		t.Error("This trigger should have passed!")
+
+	}
+
+}
+
+func TestInvalidTriggerGenericErrors(t *testing.T) {
+	DeviceTriggerGroupAndID := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		  {
+			"type": "device_trigger",
+			"on": "device_connected",
+			"device_id": "45336",
+			"group_name": "sdgfsd"
+		  }
+		]
+	  }`
+
+	_, err := ParseTriggerFrom([]byte(DeviceTriggerGroupAndID))
+	if err == nil {
+		t.Error("This trigger should have failed validation! cannot use device_id and group_name")
+	}
+
+	DeviceTriggerNoAction := `
+	{
+		"name": "test",
+		"action":"",
+		"simple_triggers": [
+		  {
+			"type": "invalid_trigger",
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerNoAction))
+	if err == nil {
+		t.Error("This trigger should have Failed! required action")
+
+	} else {
+		fmt.Println(err)
+	}
+
+	DeviceTriggerNoTriggers := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerNoTriggers))
+	if err == nil {
+		t.Error("This trigger should have Failed! invalid json match operator")
+
+	} else {
+		fmt.Println(err)
+	}
+	DeviceTriggerTooManyTriggers := `
+	{
+		"name": "test",
+		"action": {
+			"http_url": "https://example.com/my_hook",
+			"http_method": "post"
+		},
+		"simple_triggers": [
+		  {
+			"type": "device_trigger",
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  },
+		  {
+			"type": "device_trigger",
+			"on": "incoming_data",
+			"interface_name": "*",
+			"match_path":"/*",
+			"value_match_operator":"==",
+			"known_value":"3"
+		  }
+		]
+	  }`
+
+	_, err = ParseTriggerFrom([]byte(DeviceTriggerTooManyTriggers))
+	if err == nil {
+		t.Error("This trigger should have Failed! invalid json match operator")
```

Comment:
I would expect something like `"This trigger should have Failed!  Too many simple triggers found"`)